api/pkg/s3/bucketget.go

```diff
@@ -15,125 +15,176 @@
+		return
+	}
+	if listObjectsRsp.ErrorCode != http.StatusOK {
+		log.Errorf("list objects failed, ErrorCode:%d, ErrorMsg:%v\n", listObjectsRsp.ErrorCode,
+			listObjectsRsp.ErrorMsg)
+		WriteErrorResponseWithStatus(response, listObjectsRsp.ErrorCode, listObjectsRsp.ErrorMsg)
 		return
-	} else {
-		log.Infof("Get filter for BucketGet, filterOpts=%+v, filter=%+v\n",
-			filterOpts, filter)
 	}
 
-	if filter[common.KObjKey] != "" {
-		// filter[common.KObjKey] should be like: like:parttern
-		ret, err := checkObjKeyFilter(filter[common.KObjKey])
-		if err != nil {
-			log.Errorf("invalid objkey:%s\v", filter[common.KObjKey])
-			response.WriteError(http.StatusBadRequest,
-				fmt.Errorf("invalid objkey filter, it should be like objkey=like:parttern"))
+	rsp := CreateListObjectsResponse(bucketName, &req, listObjectsRsp)
+	log.Debugf("rsp:%+v\n", rsp)
+	// Write success response.
+	response.WriteEntity(rsp)
+
+	return
+}
+
+func parseListObjectsQuery(query url.Values) (request s3.ListObjectsRequest, err error) {
+	if query.Get("list-type") == constants.ListObjectsType2Str {
+		request.Version = constants.ListObjectsType2Int
+		request.ContinuationToken = query.Get("continuation-token")
+		request.StartAfter = query.Get("start-after")
+		if !utf8.ValidString(request.StartAfter) {
+			err = ErrNonUTF8Encode
 			return
 		}
-		filter[common.KObjKey] = ret
-	}
-
-	// Check validation of query parameter. Example of lastmodified: {"lt":"100", "gt":"30"}
-	if filter[common.KLastModified] != "" {
-		var tmFilter map[string]string
-		err := json.Unmarshal([]byte(filter[common.KLastModified]), &tmFilter)
-		if err != nil {
-			log.Errorf("invalid lastModified:%s\v", filter[common.KLastModified])
-			response.WriteError(http.StatusBadRequest,
-				fmt.Errorf("invalid lastmodified, it should be like lastmodified={\"lt\":\"numb\"}"))
+		request.FetchOwner = helper.Ternary(query.Get("fetch-owner") == "true",
+			true, false).(bool)
+	} else {
+		request.Version = constants.ListObjectsType1Int
+		request.Marker = query.Get("marker")
+		if !utf8.ValidString(request.Marker) {
+			err = ErrNonUTF8Encode
 			return
 		}
-		err = checkLastmodifiedFilter(&tmFilter)
+	}
+	request.Delimiter = query.Get("delimiter")
+	if !utf8.ValidString(request.Delimiter) {
+		err = ErrNonUTF8Encode
+		return
+	}
+	request.EncodingType = query.Get("encoding-type")
+	if request.EncodingType != "" && request.EncodingType != "url" {
+		err = ErrInvalidEncodingType
+		return
+	}
+	if query.Get("max-keys") == "" {
+		request.MaxKeys = utils.MaxObjectList
+	} else {
+		var maxKey int
+		maxKey, err = strconv.Atoi(query.Get("max-keys"))
 		if err != nil {
-			log.Errorf("invalid lastModified:%s\v", filter[common.KLastModified])
-			response.WriteError(http.StatusBadRequest,
-				fmt.Errorf("invalid lastmodified, it should be like lastmodified={\"lt\":\"numb\"}"))
+			log.Errorf("Error parsing max-keys:%v\n", err)
+			return request, ErrInvalidMaxKeys
```

Comment:
first letter should be lowercase in error log

Reply:
Done